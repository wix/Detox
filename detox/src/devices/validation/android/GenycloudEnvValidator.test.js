// @ts-nocheck
describe('Genymotion-cloud test environment validator', () => {

  const MOCK_GMSAAS_PATH = '/path/to/gmsaas';

  let exec;
  let uut;
  beforeEach(() => {
    jest.mock('../../../utils/environment');

    const environment = require('../../../utils/environment');
    environment.getGmsaasPath.mockReturnValue(MOCK_GMSAAS_PATH);

    const GenyExec = jest.genMockFromModule('../../allocation/drivers/android/genycloud/exec/GenyCloudExec');
    exec = new GenyExec();

    const GenycloudEnvValidator = require('./GenycloudEnvValidator');
    uut = new GenycloudEnvValidator({ exec });
  });

  const givenGmsaasExecVersion = (version) => exec.getVersion.mockResolvedValue({ version });
  const givenProperGmsaasExecVersion = () => givenGmsaasExecVersion('1.6.0');

  it('should throw an error if gmsaas exec is too old (minor version < 6)', async () => {
    givenGmsaasExecVersion('1.5.9');

    try {
      await uut.validate();
    } catch (e) {
      expect(e.constructor.name).toEqual('DetoxRuntimeError');
      expect(e.toString()).toContain(`Your Genymotion-Cloud executable (found in ${MOCK_GMSAAS_PATH}) is too old! (version 1.5.9)`);
      expect(e.toString()).toContain(`HINT: Detox requires version 1.6.0, or newer. To use 'android.genycloud' type devices, you must upgrade it, first.`);
      return;
    }
    throw new Error('Expected an error');
  });

  it('should accept the gmsaas exec if version is sufficiently new', async () => {
    givenGmsaasExecVersion('1.6.0');
    await uut.validate();
  });

  it('should accept the gmsaas exec if version is more than sufficiently new', async () => {
    givenGmsaasExecVersion('1.7.2');
    await uut.validate();
  });

  it('should throw an error if gmsaas exec is too old (major version < 1)', async () => {
    givenGmsaasExecVersion('0.6.0');

    await expect(uut.validate())
      .rejects
      .toThrowError(`Your Genymotion-Cloud executable (found in ${MOCK_GMSAAS_PATH}) is too old! (version 0.6.0)`);
  });

  it('should not throw an error if properly logged in to gmsaas', async () => {
    givenProperGmsaasExecVersion();
    await uut.validate();
  });
});
