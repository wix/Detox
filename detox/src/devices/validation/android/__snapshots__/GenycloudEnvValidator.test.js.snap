// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genymotion-cloud test environment validator health validations should throw if gmsaas doctor detects an error 1`] = `
[DetoxRuntimeError: Error: gmsaas is not configured properly
One or several issues have been detected:
- Android SDK not configured.]
`;

exports[`Genymotion-cloud test environment validator version validations should throw an error if gmsaas exec is too old (minor version < 6) 1`] = `
[DetoxRuntimeError: Your Genymotion-Cloud executable (found in /path/to/gmsaas) is too old! (version 1.5.9)

HINT: Detox requires version 1.6.0, or newer. To use 'android.genycloud' type devices, you must upgrade it, first.]
`;
